### PR TITLE
DOC: Add instruction to initialize git submodules

### PR DESCRIPTION
**Pull request by astrojuanlu**
Adding 1 missing command needed to complete the clone process, otherwise `RuntimeError: Missing `SVML` submodule! Run `git submodule update --init` to fix this.` are shown.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
I added a second commit to reference the cloning instructions from the "Building from source" procedure, otherwise it was kind of weird since no reference was made to how to obtain the actual source code. Let me know what you think.

**Comment by astrojuanlu** [comment]
cc @melissawm

File: doc/source/user/building.rst [hunk 1]
```diff
@@ -79,6 +79,10 @@ Building NumPy requires the following software installed:
 
    For building NumPy, you'll need a recent version of Cython.
 
+5) The NumPy source code
+
+   Clone the repository following the instructions in :doc:`dev/index`.
```
Comment (mattip) [inline doc/source/user/building.rst:84]: The link is not working.

Comment (astrojuanlu) [reply doc/source/user/building.rst]: One would think that, after writing the official Sphinx tutorial, I'd know better... fixed

**Review by rossbar — APPROVED** [review]
LGTM, thanks @astrojuanlu !

**Pull request merged**